contracts/chain-adapters/ForwarderBase.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+/**
+ * @title ForwarderBase
+ * @notice This contract expects to receive messages and tokens from the Hub Pool on L1 and forwards them to targets on L3. Messages
+ * are intended to originate from the re-router adapter on L1 which will send messages here to re-route them to the corresponding L3
+ * spoke pool. It rejects messages which do not originate from the cross domain admin, which should be set to the hub pool.
+ * @dev Base contract designed to be deployed on L2 to re-route messages from L1 to L3. If a message is sent to this contract which
+ * came from the L1 cross domain admin, then it is routed to the L3 via the canonical messaging bridge. Tokens sent from L1 are stored
+ * temporarily on this contract. Any EOA can initiate a bridge of these tokens to the target `l3SpokePool`.
+ * @custom:security-contact bugs@across.to
+ */
+abstract contract ForwarderBase is UUPSUpgradeable {
```

Comment:
Do you think this contract should inherit AdapterInterface?

It kind of makes sense to me that this contract should enforce that all child contracts implement relayMessage/relayTokens. All of the comments about this contract  suggest this contract should forward messages from L1 to L3, so if a child contract extended ForwarderBase but didn't extend AdapterInterface, it'd be weird

Reply:
Yeah that's a good refactor https://github.com/across-protocol/contracts/pull/609/commits/d78ad6fdd7e6b09cd7ffd4ca20d51df534ac7ba1